### PR TITLE
Ports hygiene bot

### PR DESCRIPTION
**Pull request by That0nePerson**
## About The Pull Request
Ports the hygiene bot from https://github.com/tgstation/tgstation/pull/49083. Also refactors shower code to the atom level, like that pr.

## Why It's Good For The Game
New funny bot for robo to play around with and emag.  Also less filthy people on station.

## Changelog
:cl: Qustinnus, That0nePerson
add: Adds hygiene bot! Will chase you down and shower bloody people, and can be emagged to spit fire.
/:cl:

### CODE REVIEW

**Comment by Sylphily** [comment]
Speed merge

**Comment by Raven-Industries** [comment]
Also known as the Evidence Removal bot. Keeping your getaways squeaky clean.

**Comment by FlamePrince** [comment]
The banter the bot has puts Beepsky to shame and I dunno how I feel about that. needs some starship troopers references.

also, buffing cleaning IS gonna make det's life a little harder/unimportant as evidence is just going to become a thing of the past so it'd be nice to see some balancing on the det side of things.

**Comment by That0nePerson** [comment]
janitor bot already does a much, much better job of cleaning evidence as it gets rid of the actual stuff at the crime scene

**Comment by utilbird** [comment]
oops

File: code/modules/mob/living/carbon/human/human.dm [hunk 1]
```diff
@@ -1197,6 +1197,26 @@
...
+		wear_suit.washed(washer)
+		update_inv_wear_suit()
+	else if(w_uniform)
+		w_uniform.washed(washer)
+		update_inv_w_uniform()
+
+	if(!is_mouth_covered())
+		lip_style = null
+		update_body()
+	if(belt && belt.washed(washer))
```
Comment (utilbird) [inline code/modules/mob/living/carbon/human/human.dm]:
```suggestion
	if(belt?.washed(washer))
```

File: code/modules/mob/living/carbon/human/human.dm [hunk 2]
```diff
@@ -1197,6 +1197,26 @@
...
+		update_inv_belt()
+
+	var/list/obscured = check_obscured_slots()
+
+	if(gloves && !(HIDEGLOVES in obscured) && gloves.washed(washer))
+		SEND_SIGNAL(src, COMSIG_COMPONENT_CLEAN_ACT, CLEAN_STRENGTH_BLOOD)
```
Comment (utilbird) [inline code/modules/mob/living/carbon/human/human.dm]:
```suggestion
	if(gloves && !(HIDEGLOVES in check_obscured_slots()) && gloves.washed(washer))
		SEND_SIGNAL(src, COMSIG_COMPONENT_CLEAN_ACT, CLEAN_STRENGTH_BLOOD)
```

File: code/modules/mob/living/simple_animal/bot/construction.dm [hunk 3]
```diff
@@ -520,3 +520,43 @@
...
+					to_chat(user, "<span class='warning'>You need one fluid duct to finish [src]</span>")
+					return
+				to_chat(user, "<span class='notice'>You start to pipe up [src]...</span>")
+				if(do_after(user, 40, target = src))
+					if(D.get_amount() >= 1)
+						D.use(1)
+						to_chat(user, "<span class='notice'>You pipe up [src].</span>")
+						build_step++
+					else
+						return
```
Comment (utilbird) [inline code/modules/mob/living/simple_animal/bot/construction.dm]:
```suggestion
					if(D.get_amount() <= 1)
						return
					D.use(1)
					to_chat(user, "<span class='notice'>You pipe up [src].</span>")
					build_step++
```

File: code/modules/mob/living/simple_animal/bot/hygienebot.dm [hunk 4]
```diff
@@ -0,0 +1,254 @@
...
+			target = H
+			oldtarget_name = H.name
+			speak("Unhygienic client found. Please stand still so I can clean you.")
+			playsound(loc, 'sound/effects/hygienebot_happy.ogg', 60, 1)
+			visible_message("<b>[src]</b> points at [H.name]!")
+			mode = BOT_HUNT
+			INVOKE_ASYNC(src, .proc/handle_automated_action)
+			break
+		else
+			continue
```
Comment (utilbird) [inline code/modules/mob/living/simple_animal/bot/hygienebot.dm]:
```suggestion
```

File: code/modules/mob/living/simple_animal/bot/construction.dm [hunk 5]
```diff
@@ -520,3 +520,43 @@
...
+
+/obj/item/bot_assembly/hygienebot/attackby(obj/item/I, mob/user, params)
+	. = ..()
+	switch(build_step)
+		if(ASSEMBLY_FIRST_STEP)
+			if(I.tool_behaviour == TOOL_WELDER)
+				if(I.use_tool(src, user, 0, volume=40))
+					add_overlay("hs_hole")
+					to_chat(user, "<span class='notice'>You weld a water hole in [src]!</span>")
+					build_step++
```
Comment (utilbird) [inline code/modules/mob/living/simple_animal/bot/construction.dm]:
```suggestion
			if(I.tool_behaviour == TOOL_WELDER && I.use_tool(src, user, 0, volume=40))
				add_overlay("hs_hole")
				to_chat(user, "<span class='notice'>You weld a water hole in [src]!</span>")
				build_step++
```

File: code/modules/mob/living/simple_animal/bot/construction.dm [hunk 6]
```diff
@@ -520,3 +520,43 @@
...
+
+		if(ASSEMBLY_SECOND_STEP)
+			if(!can_finish_build(I, user))
+				return
+			if(istype(I, /obj/item/stack/ducts))
+				var/obj/item/stack/ducts/D = I
+				if(D.get_amount() < 1)
+					to_chat(user, "<span class='warning'>You need one fluid duct to finish [src]</span>")
```
Comment (utilbird) [inline code/modules/mob/living/simple_animal/bot/construction.dm]:
```suggestion
					to_chat(user, "<span class='warning'>You need one fluid duct to finish [src].</span>")
```

**Review by utilbird — APPROVED** [review]

**Pull request merged**